### PR TITLE
WIP, initial commit of prometheus and grafana

### PR DESCRIPTION
**Pull request by scollier**
This PR is a WIP, but I'd like to get it merged in, then I will organize it into lab 5, and push lab 6,7,8 up one.  To test this PR, you just need a functioning OCP environment, and go through the doc here.

### CODE REVIEW

File: labs/prom-grafa.adoc [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+| link:https://instructor.labs.sysdeseng.com/rhte.ppk[https://instructor.labs.sysdeseng.com/rhte.ppk]
+| Username: student +
+Password: INSTRUCTOR WILL PROVIDE
+| Grafana
+| link:https://grafana-grafana.apps-<student_id>.labs.sysdeseng.com[grafana-grafana.apps-<student_id>.labs.sysdeseng.com]
+| Username: student +
+Password: INSTRUCTOR WILL PROVIDE
+| Prometheus
+| link:https://prometheus-openshift-metrics.apps-<student_id>.labs.sysdeseng.com[prometheus-openshift-metrics.apps-<student_id>.labs.sysdeseng.com]
+| Username: student +
```
[the remark this answers is not in the document]
Comment (scollier) [reply labs/prom-grafa.adoc]: fixed

File: labs/prom-grafa.adoc [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+| link:https://instructor.labs.sysdeseng.com/rhte.pem[https://instructor.labs.sysdeseng.com/rhte.pem]
+| Username: student +
+Password: INSTRUCTOR WILL PROVIDE
+| Windows PuTTY private key
+| link:https://instructor.labs.sysdeseng.com/rhte.ppk[https://instructor.labs.sysdeseng.com/rhte.ppk]
+| Username: student +
+Password: INSTRUCTOR WILL PROVIDE
+| Grafana
+| link:https://grafana-grafana.apps-<student_id>.labs.sysdeseng.com[grafana-grafana.apps-<student_id>.labs.sysdeseng.com]
+| Username: student +
```
[the remark this answers is not in the document]
Comment (scollier) [reply labs/prom-grafa.adoc]: fixed

File: labs/prom-grafa.adoc [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+Password: INSTRUCTOR WILL PROVIDE
+|======================
+
+== Overview
+
+In this lab, we will explore how to handle metrics and visualization of those metrics using OpenShift, link:https://prometheus.io/[Prometheus] and link:https://grafana.com/[Grafana]. We'll do a high level overview of each component, discuss how they interoperate and finally explore the environment to see how to these components together.
+
+=== Prometheus
+
+From the website: "Prometheus is an open-source systems monitoring and alerting toolkit originally built at SoundCloud.". Prometheus has gained in popularity over the past couple of years and has native integration with OpenShift which enables us to leverage it's powerful capabilities to gain insight into our clusters. Prometheus is a project in the CLoud Native Computing Foundation. As all admins and developers know, Day 1 operations of installation and configuration are fun, but the it really starts on Day 2.  Keeping an eye on our performance and making sure the cluster is functioning properly is easier with the right tools. That's where Prometheus comes in.
```
[the remark this answers is not in the document]
Comment (scollier) [reply labs/prom-grafa.adoc]: fixed

File: labs/prom-grafa.adoc [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+openshift_prometheus_args:
+  - '--storage.tsdb.retention=3d'
+  - '--web.enable-lifecycle'
+openshift_prometheus_additional_rules_file: {{ tower_openshift_prometheus_config_dir }}/prometheus_additional_rule.yml
+openshift_node_open_ports:
+  - service: "Prometheus Node Exporter"
+    port: "9100/tcp"
+openshift_prometheus_node_exporter_image_version: "{{ openshift_image_tag }}" ```
+
+Make sure you are logged in as ```system:admin```
```
[the remark this answers is not in the document]
Comment (scollier) [reply labs/prom-grafa.adoc]: fixed

File: labs/prom-grafa.adoc [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+Look at the routes, statefulsets, services and pods assocated with the openshift-metrics namespace.
+
+.master$
+[source, bash]
+----
+oc get all
+----
+
+Look at the config map as well
```
[the remark this answers is not in the document]
Comment (scollier) [reply labs/prom-grafa.adoc]: fixed

File: labs/prom-grafa.adoc [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+.master$
+[source, bash]
+----
+oc describe configmap prometheus
```
[the remark this answers is not in the document]
Comment (scollier) [reply labs/prom-grafa.adoc]: done

File: labs/prom-grafa.adoc [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+List the routes for Prometheus and open up the web interface. Copy the URL for prometheus and sign in with the user <student_id>-admin.
+
+.master$
+[source, bash]
+----
+oc get routes
+----
+
+
+- Go to web interface and add a graph manually.
```
[the remark this answers is not in the document]
Comment (scollier) [reply labs/prom-grafa.adoc]: fixed

File: labs/prom-grafa.adoc [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+
+Now that we have explored some of the capabilities of Prometheus, let's have a look at how we can use Graphana to do more visualization of the data.
+
+Clone the OpenShift Origin repo and change to the Grafana directory.
+
+.master$
+[source, bash]
+----
+cd ~
+git clone https://github.com/openshift/origin.git
```
[the remark this answers is not in the document]
Comment (scollier) [reply labs/prom-grafa.adoc]: fixed

File: labs/prom-grafa.adoc [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+.master$
+[source, bash]
+----
+oc get routes
+----
+
+
+- Go to web interface and add a graph manually.
+
+Once in the web UI, click on `Graph` in the top navigation pane.  Next, click on the `"insert metric at cursor"` dropdown and scroll to `node_memory_MemAvailable`, then click `Execute`. This should create a graph on your screen. Make sure to hover over the lines in the graph and get a sense of what's being offered here. You can continue adding graphs and organizing your dashboard how you want it.
```
[the remark this answers is not in the document]
Comment (scollier) [reply labs/prom-grafa.adoc]: fixed

File: labs/prom-grafa.adoc [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+
+Once you have created a graph or two, take some time to explore the other options on the top navigation bar: `Alerts`, `Status`, `Help`. In the next section, we are going to spend some time on `Alerts`.
```
[the remark this answers is not in the document]
Comment (scollier) [reply labs/prom-grafa.adoc]: fixed

File: labs/prom-grafa.adoc (hunk 8 above)
[the remark this answers is not in the document]
Comment (scollier) [reply labs/prom-grafa.adoc]: fixed

File: labs/prom-grafa.adoc [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+[source, bash]
+----
+./setup-grafana.sh -n prometheus -p openshift-metrics
+oc project grafana
+oc get routes
+----
+
+Log in with your OpenShift admin account: <student_id>-admin and the password from your instructor.
+
+grafana-grafana.apps-<student_id>.labs.sysdeseng.com
```
[the remark this answers is not in the document]
Comment (scollier) [reply labs/prom-grafa.adoc]: fixed

File: labs/prom-grafa.adoc [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+
+In the upper left navigation pane, click the `Home` button and that will expose the `openshift cluster monitoring` link.  Click that.
```
[the remark this answers is not in the document]
Comment (scollier) [reply labs/prom-grafa.adoc]: tried.

File: labs/prom-grafa.adoc [hunk 13]
```diff
@@ -0,0 +1,165 @@
...
+
+.master$
+[source, bash]
+----
+cd ~
+git clone https://github.com/openshift/origin.git
+cd origin/examples/grafana/
+----
+
+Deploy Grafana and grab the route.
```
[the remark this answers is not in the document]
Comment (scollier) [reply labs/prom-grafa.adoc]: i def had it framed out like that at first.  fixed.

File: labs/prom-grafa.adoc [hunk 14]
```diff
@@ -111,27 +110,32 @@ oc get routes
 ----
 
 
-- Go to web interface and add a graph manually.
+- Go to web interface and add a graph manually. The credentials are listed in the table at the top of the page. Your route will look something like:
 
-Once in the web UI, click on `Graph` in the top navigation pane.  Next, click on the `"insert metric at cursor"` dropdown and scroll to `node_memory_MemAvailable`, then click `Execute`. This should create a graph on your screen. Make sure to hover over the lines in the graph and get a sense of what's being offered here. You can continue adding graphs and organizing your dashboard how you want it.
+```
+prometheus-openshift-metrics.apps-<student_id>.labs.sysdeseng.com
```
[the remark this answers is not in the document]
Comment (sabre1041) [reply labs/prom-grafa.adoc]: @scollier please submit subsequent PR with this change

**Comment by sabre1041** [comment]
Merging to keep things moving

**Pull request merged**